### PR TITLE
add options to create_scheduler so that the get_runner method is full…

### PR DESCRIPTION
**Pull request by KPostOffice**
…y configurable

I also added a note to each schedulers __init__ method to help with maintainablility

I changed the create scheduler function for the local scheduler to specify the key word args that it takes rather than getting them from the `**kwargs` dict.

Test plan:
- [ ] add unit tests

### CODE REVIEW

File: torchx/schedulers/ray_scheduler.py [hunk 1]
```diff
@@ -441,10 +442,12 @@ def list(self) -> List[ListAppResponse]:
...
+def create_scheduler(
+    session_name: str, ray_client: Optional[JobSubmissionClient] = None, **kwargs: Any
+) -> "RayScheduler":
     if not has_ray():  # pragma: no cover
         raise ModuleNotFoundError(
             "Ray is not installed in the current Python environment."
         )
 
-    return RayScheduler(session_name=session_name)
+    return RayScheduler(session_name=session_name, ray_client=ray_client)
```
Comment (d4l3k) [inline torchx/schedulers/ray_scheduler.py:453]: would be nice to add some unit tests for this

**Review by d4l3k — APPROVED** [review]
LGTM -- thanks! A unit test would be nice but not critical and fixing lint

**Comment by KPostOffice** [comment]
@d4l3k I've added those unit tests. Feel free to merge if you think the PR is good to go

**Review by d4l3k — APPROVED** [review]
LGTM thanks for adding!

**Pull request merged**